server/sonar-server/src/main/java/org/sonar/server/authentication/event/AuthenticationEvent.java

```diff
@@ -30,17 +32,21 @@
 
   void login(HttpServletRequest request, String login, Source source);
 
+  void failure(HttpServletRequest request, AuthenticationException e);
+
   enum Method {
-    BASIC, BASIC_TOKEN, FORM, FORM_TOKEN, SSO, OAUTH2, EXTERNAL
+    BASIC, BASIC_TOKEN, FORM, FORM_TOKEN, SSO, OAUTH2, JWT, CSRF, EXTERNAL
```

Comment:
As discussed, CSFR is not an authentication method.
CSRF failure should be either catched in OAUTH or in JWT.